pkg/volume/util/util.go

```diff
@@ -571,6 +571,34 @@ func IsLocalEphemeralVolume(volume v1.Volume) bool {
 		volume.ConfigMap != nil
 }
 
+// GetLocalLocalPersistentVolumeNodeName determines whether the argument is a local PersistentVolume
+// and return the node name for the PV node affinity if it is a local PV.
+// Local PV has node affinity in the form of:
+// nodeAffinity:
+//	required:
+//	  nodeSelectorTerms:
+//	  - matchExpressions:
+//		- key: kubernetes.io/hostname
+//		  operator: In
+//		  values:
+//		  - <node>
+func GetLocalLocalPersistentVolumeNodeName(pv *v1.PersistentVolume) (string, bool) {
+	if pv != nil && pv.Spec.NodeAffinity != nil && pv.Spec.NodeAffinity.Required != nil {
```

Comment:
stylish comment: the 5 ifs make the code intended too much and hard to read. I'd like code it like:

```go
if pv == nil || pv.Spec.NodeAffinity == nil || pv.Spec.NodeAffinity.Required == nil {
    return
}
if len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) != 1 {
    return
}
......

```

Reply:
Sure thing, restructured to use negative checks and return early style.